### PR TITLE
Use clang-format for code style

### LINKED ISSUE #186: clang-format (or something...)?
Over the years it seems this codebase has accrued both technical debt and style debt. What's done is done, but going forward it would be nice to help new contributors by not making them just guess whatever particular style or format a specific file has.

Some things I've noticed that change file to file:
- Indentation is all over the place
- Mixing tabs and spaces
- Braces being on the same line or after a new line (both for functions and conditionals/loops, both different)
- Spacing between lines
- Spacing between arguments and operators

I'm not too sure how to handle something like this or the kind of code churn that something like `clang-format` will bring.

### PR DESCRIPTION
**Pull request by LaserEyess**
This is highly opinionated, but it is a style that tries to reduce the
amount of churn in the code base. There are many places where whole
files need to be re-written because their styles are incredibly outside
of every other file.

The CI is updated with a format checker as well as an included
scripts/pre-commit git hook for local use.

closes #186

### CODE REVIEW

**Comment by LaserEyess** [comment]
How in the world did the formatting fail after I reformatted it.

File: src/bencode.c [hunk 1]
```diff
@@ -26,45 +26,45 @@
...
 
 #include <glib.h>
 
 #include "bencode.h"
 
-static be_node *be_alloc(be_type type)
+static be_node* be_alloc(be_type type)
 {
-    be_node *ret = g_malloc0(sizeof(be_node));
+    be_node* ret = g_malloc0(sizeof(be_node));
```
Comment (TingPing) [inline src/bencode.c:39]: This style is very uncommon in C codebases, none within the glib ecosystem. It's more of a C++ thing IME.

**Comment by TingPing** [comment]
Outside of the one opinion I think it seems pretty reasonable.

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿---
+BasedOnStyle: Webkit
```
Comment (TingPing) [inline .clang-format:2]: Caee sensitive, WebKit?

**Comment by LaserEyess** [comment]
I really don't understand this, locally I have:
```
$ clang-format --version
clang-format version 13.0.1
```

The same version in this CI action I found, and yet there is some different in `BreakBeforeBraces`?

**Comment by LaserEyess** [comment]
Ok, I figured it out. `clang-format` 13 has a bug I guess, the solution is moving to 14. 

@TingPing if you have any other opinions I'd definitely be willing to hear. This style was just me looking at all of the options and thinking to myself "this looks good". No real reason beyond that.

**Pull request merged**